src/app/lists/Dataview.tsx

```diff
@@ -0,0 +1,195 @@
+    ["7_quantity", "Family of 7"],
+    ["8_quantity", "Family of 8"],
+    ["9_quantity", "Family of 9"],
+    ["10_quantity", "Family of 10+"],
+];
+
+export const tooltips: [string, string][] = [
+    ["1_notes", "Single"],
+    ["2_notes", "Family of 2"],
+    ["3_notes", "Family of 3"],
+    ["4_notes", "Family of 4"],
+    ["5_notes", "Family of 5"],
+    ["6_notes", "Family of 6"],
+    ["7_notes", "Family of 7"],
+    ["8_notes", "Family of 8"],
+    ["9_notes", "Family of 9"],
+    ["10_notes", "Family of 10+"],
+];
+
+const ListsDataView: React.FC<Props> = ({ data: rawData }) => {
+    // null => add, undefined => modal closed
+    const [modal, setModal] = useState<Datum | null | undefined>(undefined);
+    const [toDelete, setToDelete] = useState<number | null>(null);
+    // need another setState otherwise the modal content changes before the close animation finishes
+    const [toDeleteModalOpen, setToDeleteModalOpen] = useState<boolean>(false);
+    const [errorMsg, setErrorMsg] = useState<string | null>(null);
+
+    if (rawData === null) {
+        throw new Error("No data found");
+    }
+
+    const remapTooltips = (row: ListRow): Datum & { unmappedTooltips: RowData } => {
+        const data: RowData = {
+            item_name: row.item_name,
+            primary_key: row.primary_key,
+        };
+        const tooltips: RowData = {};
+        const unmappedTooltips: RowData = {};
+
+        for (const [key, value] of Object.entries(row)) {
+            if (key.endsWith("quantity") && value) {
+                data[key] = value;
+            } else if (key.endsWith("notes") && value) {
+                tooltips[key.replace("notes", "quantity")] = value;
+                unmappedTooltips[key] = value;
+            }
+        }
+
+        return {
+            data,
+            tooltips,
+            unmappedTooltips,
+        };
+    };
+
+    const dataAndTooltips = rawData?.map(remapTooltips);
+
+    // extract header keys
+    const toggleableHeaders = headers.map(([key]) => key);
+    // removw description header
+    toggleableHeaders.shift();
+
+    const onEdit = (index: number): void => {
+        const row = rawData[index];
+
+        const editData = remapTooltips(row!);
+
+        setModal({
+            data: editData.data,
+            tooltips: editData.unmappedTooltips,
+        });
+    };
+
+    const onDeleteButtonClick = (index: number): void => {
+        setToDelete(index);
+        setToDeleteModalOpen(true);
+    };
+
+    const onConfirmDeletion = async (): Promise<void> => {
+        if (toDelete !== null) {
+            const data = rawData[toDelete];
+            const response = await supabase
+                .from("lists")
+                .delete()
+                .match({ primary_key: data.primary_key });
+
+            // this is not a normal js response object, so checking response.error (and using response.error.message) is fine
+            if (response.error) {
```

Comment:
If you're not using the rest of the `response` (i.e. `data`, `status`, ...) you can just do `const { error } = ...` on line 107 instead :)

Reply:
I could, although I feel that this would harm readability